### PR TITLE
gh-eco: init 0.1.0

### PR DESCRIPTION
**Pull request by helium18**
###### Description of changes
Add a package by which one can "explore the GitHub ecosystem"

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by fufexan — APPROVED** [review]
Looks good, but wasn't `buildGoPackage` deprecated in favor of `buildGoModule`?

**Comment by helium18** [comment]
thanks for pointing that out, I wasn't aware of that

**Review by fufexan — APPROVED** [review]

File: pkgs/tools/misc/gh-eco/default.nix [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  ];
+
+  # No tests
+  doCheck = false;
+
+  meta = with lib; {
+    homepage = "https://github.com/coloradocolby/gh-eco";
+    description = "gh extension to explore the ecosystem";
+    license = licenses.mit;
+    platforms = go.meta.platforms or lib.platforms.all;
```
Comment (bobby285271) [inline pkgs/tools/misc/gh-eco/default.nix:34]: Default for `buildGoModule`

```suggestion
```

File: pkgs/tools/misc/gh-eco/default.nix [hunk 2]
```diff
@@ -0,0 +1,38 @@
+{ lib
+, go
```
Comment (bobby285271) [inline pkgs/tools/misc/gh-eco/default.nix:2]:
```suggestion
```

File: pkgs/tools/misc/gh-eco/default.nix [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    owner = "coloradocolby";
+    repo = "gh-eco";
+    rev = "v${version}";
+    sha256 = "sha256-rJG1k8lOyXQSP3FgyyHZvVrQkn2yEtAcgg9CpbPvCwY=";
+  };
+
+  vendorSha256 = "sha256-Qx/QGIurjKGFcIdCot1MFPatbGHfum48JOoHlvqA64c=";
+
+  goDeps = ./deps.nix;
+
```
Comment (bobby285271) [inline pkgs/tools/misc/gh-eco/default.nix:21]: Not useful for `buildGoModule`

```suggestion
```

File: pkgs/tools/misc/gh-eco/deps.nix [hunk 4]
```diff
@@ -0,0 +1,453 @@
+# file generated from go.mod using vgo2nix (https://github.com/nix-community/vgo2nix)
```
Comment (bobby285271) [inline pkgs/tools/misc/gh-eco/deps.nix:1]: We can drop this file.

**Review by bobby285271 — COMMENTED** [review]
Welcome! Please squash the commits into exactly two:

- `maintainers: add helium` for the `maintainers/maintainer-list.nix` edit
- `gh-eco: init at 0.1.0` for the package addition

**Comment by helium18** [comment]
Thanks @bobby285271 for the suggestions. I'll probably fix this now
So to confirm, I have to 
1. Reflect the suggestions in a new commit 
2. Squash all the commits into two, i.e 
- `maintainers: add helium`
- `gh-eco: init 0.1.0`

**Comment by 06kellyjac** [comment]
>     1. Reflect the suggestions in a new commit
> 
>     2. Squash all the commits into two, i.e
> 
> 
>     * `maintainers: add helium`
> 
>     * `gh-eco: init 0.1.0`

exactly

(also please run `nixpkgs-fmt` on it :) )

**Comment by 06kellyjac** [comment]
> I've incorporated all the suggestions. I'm pushing the changes (sry this is my first time and im a bit anxious sweat )

No pressure at all. We're almost there :)

---

a couple notes:

`maintainers: add helium18` should be `maintainers: add helium` if the attribute its named `helium`

https://github.com/NixOS/nixpkgs/pull/175699/files#diff-c1fcec3afa1da96245de1d7954ca721c979a5f25aee0e8b59d59d3d21e31f8faR5077

the deps file that's unused is still there and needs deleting

`pkgs/tools/misc/gh-eco/default.nix` can still benefit from running `nixpkgs-fmt`

**Comment by helium18** [comment]
@06kellyjac did it :+1:

**Review by 06kellyjac — APPROVED** [review]
Nice.

My only remaining question is why add `doCheck = false;`

1. there seems to be a test https://github.com/coloradocolby/gh-eco/blob/v0.1.0/utils/utils_test.go
2. if there aren't tests then in the future they can run when the package is updated, I'd just leave it unset unless there's a specific reason not to run them

**Comment by helium18** [comment]
Well, I was referring to the source of other go modules like [skate](https://github.com/NixOS/nixpkgs/blob/nixos-unstable/pkgs/applications/misc/skate/default.nix#L21) and [glow](https://github.com/NixOS/nixpkgs/blob/nixos-21.11/pkgs/applications/editors/glow/default.nix#L21) for example (both of which are built with [charm.sh](https://charm.sh/) and gh-eco is built with charm.sh as well), therefore I assumed that they needed this flag

I'll remove it if that's the case :+1:

**Review by bobby285271 — APPROVED** [review]

**Pull request merged**